src/js/toolbar.js

```diff
@@ -323,25 +319,23 @@ var Toolbar;
                 if (!selectionElement ||
                         this.base.elements.indexOf(selectionElement) === -1 ||
                         selectionElement.getAttribute('data-disable-toolbar')) {
-                    this.hideToolbar();
-                    return;
+                    return this.hideToolbar();
                 }
 
                 // Now we know there's a focused editable with a selection
 
                 // If the updateOnEmptySelection option is true, show the toolbar
                 if (this.options.updateOnEmptySelection && this.options.staticToolbar) {
-                    this.showAndUpdateToolbar();
-                    return;
+                    return this.showAndUpdateToolbar();
                 }
 
                 // If we don't have a 'valid' selection -> hide toolbar
                 if (this.options.contentWindow.getSelection().toString().trim() === '' ||
                     (this.options.allowMultiParagraphSelection === false && this.multipleBlockElementsSelected())) {
-                    this.hideToolbar();
-                } else {
-                    this.showAndUpdateToolbar();
+                    return this.hideToolbar();
                 }
+
+                this.showAndUpdateToolbar();
```

Comment:
Seems like we should probably `return this.showAndUpdateToolbar()` here as well, no?